libs/core/langchain_core/prompts/base.py

```diff
@@ -105,9 +107,14 @@ def get_input_schema(
         self, config: Optional[RunnableConfig] = None
     ) -> Type[BaseModel]:
         # This is correct, but pydantic typings/mypy don't think so.
+        required_input_variables = {
+            k: (self.input_types.get(k, str), ...) for k in self.input_variables
```

Comment:
This seems to adopt the convention that `input_variables` represent required variables rather than all possible inputs, which I think is inconsistent with the behavior in the message history?

Reply:
No, it is consistent. If the message history is declared as optional, then it will only be in `optional_variables` and not in `required_input_variables`. However, if the message history is not declared as optional, then it will be in `input_variables` and therefore in `required_input_variables`. `input_variables` will always show the variables that are REQUIRED in the prompt.